### PR TITLE
Merge static libs and bump version 2.2-3

### PR DESCRIPTION
**Pull request by ElliotScher**
WPIcal had linker errors related to the Ceres dependency.

### CODE REVIEW

File: publish.gradle [hunk 1]
```diff
@@ -114,7 +117,107 @@ task cppHeadersZip(type: Zip) {
...
+task mergeStaticLibs(type: Exec) {
+    def buildDirectory = Paths.get(projectDir.canonicalPath).resolve("vcpkg_installed").resolve(triplet)
+
+    if (project.platformPath.startsWith('windows')) {
+        workingDir buildDirectory.resolve("lib").toString()
+        executable 'lib'
+        def inputFiles = ["ceres.lib", "gflags.lib", "GKlib.lib", "glog.lib", "lapack.lib", "libamd.lib", "libbtf.lib", "libcamd.lib", "libccolamd.lib", "libcholmod.lib", "libcolamd.lib", "libcxsparse.lib", "libklu.lib", "libldl.lib", "libspqr.lib", "libumfpack.lib", "metis.lib", "openblas.lib", "suitesparseconfig.lib"]
+        if (project.platformPath.contains('arm64')) {
+            print("ARM64 Architecture Detected")
+            inputFiles = ["ceres.lib", "gflags.lib", "GKlib.lib", "glog.lib", "lapack.lib", "libamd.lib", "libbtf.lib", "libcamd.lib", "libccolamd.lib", "libcholmod.lib", "libcolamd.lib", "libcxsparse.lib", "libklu.lib", "libldl.lib", "libspqr.lib", "libumfpack.lib", "metis.lib", "openblas.lib", "suitesparseconfig.lib", "libf2c.lib"]
```
Comment (Gold856) [inline publish.gradle]:
```suggestion
            inputFiles = inputFiles + "libf2c.lib"
```

File: publish.gradle [hunk 2]
```diff
@@ -114,7 +117,107 @@ task cppHeadersZip(type: Zip) {
...
+task mergeDebugStaticLibs(type: Exec) {
+    def buildDirectory = Paths.get(projectDir.canonicalPath).resolve("vcpkg_installed").resolve(triplet).resolve("debug")
+
+    if (project.platformPath.startsWith('windows')) {
+        workingDir buildDirectory.resolve("lib").toString()
+        executable 'lib'
+        def inputFiles = ["ceres-debug.lib", "gflags_debug.lib", "GKlib.lib", "glog.lib", "lapack.lib", "libamdd.lib", "libbtfd.lib", "libcamdd.lib", "libccolamdd.lib", "libcholmodd.lib", "libcolamdd.lib", "libcxsparsed.lib", "libklud.lib", "libldld.lib", "libspqrd.lib", "libumfpackd.lib", "metis.lib", "openblas.lib", "suitesparseconfigd.lib"]
+        if (project.platformPath.contains('arm64')) {
+            print("ARM64 Architecture Detected")
+            inputFiles = ["ceres-debug.lib", "gflags_debug.lib", "GKlib.lib", "glog.lib", "lapack.lib", "libamdd.lib", "libbtfd.lib", "libcamdd.lib", "libccolamdd.lib", "libcholmodd.lib", "libcolamdd.lib", "libcxsparsed.lib", "libf2c.lib", "libklud.lib", "libldld.lib", "libspqrd.lib", "libumfpackd.lib", "metis.lib", "openblas.lib", "suitesparseconfigd.lib"]
```
Comment (Gold856) [inline publish.gradle]:
```suggestion
            inputFiles = inputFiles + "libf2c.lib"
```

**Review by Gold856 — CHANGES_REQUESTED** [review]

File: publish.gradle [hunk 3]
```diff
@@ -58,7 +61,7 @@ ext.platformPath = getPlatformPath(triplet)
 ext.platformClassifier = ext.platformPath.replaceFirst('/', '')
 
 ext.repo = "thirdparty-ceres"
-def pubVersion = "2.2-2"
+def pubVersion = "2.2-3"
```
Comment (Gold856) [inline publish.gradle]:
```suggestion
def pubVersion = "$ext.version-3"
```

**Review by Gold856 — CHANGES_REQUESTED** [review]
#9 will be changing how the x86 Windows libraries are built to match ARM Windows. As far I can tell, the only changes you need to make are including libf2c into the base inputFiles array and using `gflags_static` and `gflags_static_debug` instead of just `gflags` and `gflags_debug`.

Also, consider specifying the file extension when you iterate over inputFiles instead of specifying it for all the individual libraries. I think it's more readable with less file extensions.

**Comment by ElliotScher** [comment]
are ```gflags_static``` and ```gflags_static_debug``` the libs for windows? on my linux machine they are ```gflags``` and ```gflags_debug```

**Comment by Gold856** [comment]
Yes.
image

**Pull request merged**